common/src/test/java/io/netty/util/concurrent/DefaultThreadFactoryTest.java

```diff
@@ -0,0 +1,262 @@
+                }
+            };
+
+            final AtomicReference<Throwable> interrupted = new AtomicReference<Throwable>();
+
+            // create the thread factory, since we are running the thread group brother, the thread
+            // factory will now forever be tied to that group
+            // we then create a thread from the factory to run a "task" for us
+            final Thread first = new Thread(new ThreadGroup("brother"), new Runnable() {
+                @Override
+                public void run() {
+                    factory.set(new DefaultThreadFactory("test", false, Thread.NORM_PRIORITY, null));
+                    final Thread t = factory.get().newThread(task);
+                    t.start();
+                    try {
+                        t.join();
+                    } catch (InterruptedException e) {
+                        interrupted.set(e);
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            });
+            first.start();
+            first.join();
+
+            assertNull(interrupted.get());
+
+            // now we will use factory again, this time from a sibling thread group sister
+            // if DefaultThreadFactory is "sticky" about thread groups, a security manager
+            // that forbids sibling thread groups from messing with each other will strike this down
+            final Thread second = new Thread(new ThreadGroup("sister"), new Runnable() {
+                @Override
+                public void run() {
+                    final Thread t = factory.get().newThread(task);
+                    t.start();
+                    try {
+                        t.join();
+                    } catch (InterruptedException e) {
+                        interrupted.set(e);
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            });
+            second.start();
+            second.join();
+
+            assertNull(interrupted.get());
+
+            assertThat(counter.get(), equalTo(2));
+        } finally {
+            System.setSecurityManager(current);
+        }
+    }
+
+    // test that when DefaultThreadFactory is constructed with a sticky thread group, threads
+    // created by it have the sticky thread group
+    @Test(timeout = 2000)
+    public void testDefaultThreadFactoryStickyThreadGroupConstructor() throws InterruptedException {
+        final ThreadGroup sticky = new ThreadGroup("sticky");
+        runStickyThreadGroupTest(new Callable<DefaultThreadFactory>() {
+                                     @Override
+                                     public DefaultThreadFactory call() throws Exception {
+                                         return new DefaultThreadFactory("test", false, Thread.NORM_PRIORITY, sticky);
+                                     }
+                                 },
+                sticky);
```

Comment:
Yeah, I found it strange too but that's what IntelliJ did to it. How would you like it formatted?